### PR TITLE
Adds a document about the fakes extension in MSTest Runner

### PR DESCRIPTION
**Pull request by drognanar**
Explains what the Fakes extension does, points to the MSLearn page about the fakes extension, specifies the minimum supported VS version and also specifies how to upgrade an existing project to use the extension with MSTestRunner.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/testing/unit-testing-platform-extensions-fakes.md](https://github.com/dotnet/docs/blob/117eb9b56623eb6b97096eba15eb1335a71688ed/docs/core/testing/unit-testing-platform-extensions-fakes.md) | [Fakes extension](https://review.learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-extensions-fakes?branch=pr-en-us-41517) |
| [docs/core/testing/unit-testing-platform-extensions.md](https://github.com/dotnet/docs/blob/117eb9b56623eb6b97096eba15eb1335a71688ed/docs/core/testing/unit-testing-platform-extensions.md) | [Microsoft.Testing.Platform extensions](https://review.learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-extensions?branch=pr-en-us-41517) |

### CODE REVIEW

File: docs/core/testing/unit-testing-platform-extensions-fakes.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+title: Microsoft.Testing.Platform Fakes extension
+description: Learn about the various Microsoft.Testing.Platform Fakes extension and how to use it.
+author: drognanar
+ms.author: arturs
+ms.date: 06/21/2024
+---
+
+# Fakes extension
+
+This extension provides suppoort to execute a test project that makes use of `Microsoft Fakes`.
```
Comment (Evangelink) [inline docs/core/testing/unit-testing-platform-extensions-fakes.md:11]:
```suggestion
This extension provides support to execute a test project that makes use of `Microsoft Fakes`.
```

File: docs/core/testing/unit-testing-platform-extensions-fakes.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+[Microsoft Fakes](https://learn.microsoft.com/en-us/visualstudio/test/isolating-code-under-test-with-microsoft-fakes) allows you to better test your code by either generating `Stub`s (for instance creating a testable implementation of `INotifyPropertyChanged`) or by `Shim`ing methods and static methods (replacing the implementation of `File.Open` with a one you can control in your tests).
```
Comment (Evangelink) [inline docs/core/testing/unit-testing-platform-extensions-fakes.md:13]: You should be using relative links to the md file.

File: docs/core/testing/unit-testing-platform-extensions-fakes.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+> [!NOTE]
+> This extension requires a VS Enterprise installation with the minimum version of 17.11 preview 1 in order to work correctly.
+
+# Upgrading your project to the new extension
```
Comment (Evangelink) [inline docs/core/testing/unit-testing-platform-extensions-fakes.md:18]: Only one header should be using top level header

File: docs/core/testing/unit-testing-platform-extensions-fakes.md (hunk 2 above)
Comment (drognanar) [reply docs/core/testing/unit-testing-platform-extensions-fakes.md:13]: the thing is that I could not find the fakes documentation in this repository.

there's also the `aka.ms/msfakes` shortlink that could be used instead.

File: docs/core/testing/unit-testing-platform-extensions-fakes.md (hunk 1 above)
Comment (drognanar) [reply docs/core/testing/unit-testing-platform-extensions-fakes.md]: fixed

File: docs/core/testing/unit-testing-platform-extensions-fakes.md (hunk 3 above)
Comment (drognanar) [reply docs/core/testing/unit-testing-platform-extensions-fakes.md]: fixed

**Review by Evangelink — APPROVED** [review]
@IEvangelist could you please review?

File: docs/core/testing/unit-testing-platform-extensions-fakes.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+---
+title: Microsoft.Testing.Platform Fakes extension
+description: Learn about the various Microsoft.Testing.Platform Fakes extension and how to use it.
```
Comment (IEvangelist) [inline docs/core/testing/unit-testing-platform-extensions-fakes.md:3]:
```suggestion
description: Learn about the various Microsoft.Testing.Platform Fakes extension capabilities and how to use it.
```

File: docs/core/testing/unit-testing-platform-extensions-fakes.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+author: drognanar
+ms.author: arturs
+ms.date: 06/21/2024
+---
+
+# Fakes extension
+
+This extension provides support to execute a test project that makes use of `Microsoft Fakes`.
```
Comment (IEvangelist) [inline docs/core/testing/unit-testing-platform-extensions-fakes.md:11]:
```suggestion
The `Microsoft.Testing.Extensions.Fakes` extension provides support to execute a test project that makes use of `Microsoft Fakes`.
```

**Review by IEvangelist — APPROVED** [review]

**Pull request merged**